### PR TITLE
refactor(propagation): Make Propagator all static

### PR DESCRIPTION
**Pull request by P403n1x87**
## Description

All the current implementations of the Propagator abstract class could
be made to have static methods. This avoids having to create objects
that are used just to invoke a method which could be called on the class
itself.

Only drawback is if, in the future, the inject and extract method would
depend on some internal state of the Propagator object.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] ~~Library documentation is updated.~~
- [ ] ~~[Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).~~

### CODE REVIEW

**Review by majorgreys — APPROVED** [review]

File: ddtrace/contrib/botocore/patch.py [hunk 1]
```diff
@@ -27,99 +27,92 @@
 # Original botocore client class
 _Botocore_client = botocore.client.BaseClient
 
-ARGS_NAME = ('action', 'params', 'path', 'verb')
-TRACED_ARGS = ['params', 'path', 'verb']
+ARGS_NAME = ("action", "params", "path", "verb")
```
Comment (jd) [inline ddtrace/contrib/botocore/patch.py:30]: It looks like you ran black on this but did not remove it from the exclude list in `pyproject.yaml`, right?

**Review by jd — CHANGES_REQUESTED** [review]

File: ddtrace/contrib/botocore/patch.py (hunk 1 above)
Comment (P403n1x87) [reply ddtrace/contrib/botocore/patch.py:30]: Now that you point that out I have realised that there are exclusions in `pyproject.toml`. I have black set to run on save in VS Code. Is there a reason why `botocore` is excluded from black?

Comment (brettlangdon) [reply ddtrace/contrib/botocore/patch.py:30]: We haven't gotten to formatting it yet. We went for a deny list approach to ensure all new files added are formatted, and we can slowly update all files to eventually be formatted.

If you want, you can submit a separate PR for formatting `botocore` and removing it from `pyproject.toml` exclusion.

File: ddtrace/propagation/http.py [hunk 2]
```diff
@@ -100,15 +101,16 @@ def extract_origin(headers):
...
         """Extract a Context from HTTP headers into a new Context.
 
         Here is an example from a web endpoint::
 
             from ddtrace.propagation.http import HTTPPropagator
 
             def my_controller(url, headers):
-                propagator = HTTPPropagator()
+                propagator = HTTPPropagator
                 context = propagator.extract(headers)
```
Comment (Kyle-Verhoog) [inline ddtrace/propagation/http.py:114]:
```suggestion
                context = HTTPPropagator.extract(headers)
```

File: ddtrace/propagation/http.py [hunk 3]
```diff
@@ -100,15 +101,16 @@ def extract_origin(headers):
...
+    def extract(headers):
         """Extract a Context from HTTP headers into a new Context.
 
         Here is an example from a web endpoint::
 
             from ddtrace.propagation.http import HTTPPropagator
 
             def my_controller(url, headers):
-                propagator = HTTPPropagator()
+                propagator = HTTPPropagator
```
Comment (Kyle-Verhoog) [inline ddtrace/propagation/http.py:113]:
```suggestion
```

File: ddtrace/contrib/kombu/patch.py [hunk 4]
```diff
@@ -23,11 +23,14 @@
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/kombu/patch.py:23]: Let's `black` this file in a separate PR

**Review by Kyle-Verhoog — COMMENTED** [review]
Looks good, just need to `black` celery, kombu and others first.

**Review by jd — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**